### PR TITLE
   Initial check-in. More routines are ready. This is just the first for the first test

### PR DESCRIPTION
**Pull request by gregmhenry**

### CODE REVIEW

File: samples/encoder/compile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+rm -rf xed
```
Comment (alheinecke) [inline samples/encoder/compile:1]: Please add default execution env sh/bash

File: samples/encoder/compile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+rm -rf libxed*.a
+rm -rf mbuild
+rm -rf objs/convert_int_to_name-c.o objs/encoder_x86_decode_with_xed_test-c.o
+git clone https://github.com/intelxed/xed.git xed
+git clone https://github.com/intelxed/mbuild.git mbuild
+cd xed
+./mfile.py install
```
Comment (alheinecke) [inline samples/encoder/compile:8]: please add versioned binary for python

File: samples/encoder/compile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+mv kits/xed-install-base* kits/xed-install-base
+cp -pr kits/xed-install-base/lib/libxed*.a ..
+cd ..
+icc -DNDEBUG -DLIBXSMM_OPENMP_SIMD  -I. -I../../include -I../../src -fPIC -g -Wall -diag-disable 1879,3415,3948,10006,10010,10411,13003 -O2 -qno-offload -fno-alias -ansi-alias -fopenmp -qopenmp-simd -funroll-loops -fdata-sections -ffunction-sections -fvisibility=hidden -pthread -msse4.2 -c convert_int_to_name.c -o obj/convert_int_to_name-c.o
+icc -DNDEBUG -DLIBXSMM_OPENMP_SIMD -DXED_DECODE_TESTING -I. -I../../include -I../../src -I./xed/kits/xed-install-base/include -I./xed/kits/xed-install-base/examples -fPIC -g -Wall -diag-disable 1879,3415,3948,10006,10010,10411,13003 -O2 -qno-offload -fno-alias -ansi-alias -fopenmp -qopenmp-simd -funroll-loops -fdata-sections -ffunction-sections -fvisibility=hidden -pthread -msse4.2 -c encoder_x86.c -o obj/encoder_x86_decode_with_xed_test-c.o
```
Comment (alheinecke) [inline samples/encoder/compile:13]: this belongs into the libxsmm makefile

File: samples/encoder/convert_int_to_name.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#include <stdio.h>
```
Comment (alheinecke) [inline samples/encoder/convert_int_to_name.c:1]: This entire file should be removed, please use this existing routine instead: https://github.com/hfp/libxsmm/blob/master/src/generator_common.h#L1032

File: samples/encoder/encoder_x86.c [hunk 5]
```diff
@@ -14,6 +14,363 @@
 
 #include <generator_x86_instructions.h>
 
+/* #define XED_DECODE_TESTING */
```
Comment (alheinecke) [inline samples/encoder/encoder_x86.c:17]: Can you please move these routines into a file such that all XED related routines are in this file

**Review by alheinecke — CHANGES_REQUESTED** [review]

**Comment by hfp** [comment]
Shall we close this PR?

**Comment by alheinecke** [comment]
Do we have a timeline on this?

**Pull request closed without merging**